src/hooks/useBasicTokens.ts

```diff
@@ -0,0 +1,172 @@
+              [address]: pool.typeOfAsset,
+            })),
+          ) as Record<string, PoolTypes>
+          return tokensInPool
+        })
+        .flat(),
+    )
+    if (minichefData) {
+      // add minichef reward tokens
+      minichefData.allRewardTokens.forEach((address) => {
+        targetTokenAddresses.add(address)
+      })
+    }
+    if (gauges) {
+      // add gauge tokens
+      Object.values(gauges).forEach((gauge) => {
+        gauge.rewards.forEach(({ tokenAddress }) => {
+          targetTokenAddresses.add(tokenAddress)
+        })
+      })
+    }
+    if (SDL_WETH_SUSHI_LP_CONTRACT_ADDRESSES[chainId] && gaugesAreActive) {
+      // add sushi token
+      targetTokenAddresses.add(
+        SDL_WETH_SUSHI_LP_CONTRACT_ADDRESSES[chainId].toLowerCase(),
+      )
+    }
+    if (VOTING_ESCROW_CONTRACT_ADDRESS[chainId] && gaugesAreActive) {
+      // add voting escrow token
+      targetTokenAddresses.add(
+        VOTING_ESCROW_CONTRACT_ADDRESS[chainId].toLowerCase(),
+      )
+    }
+    const tokenInfos = await getTokenInfos(
+      ethCallProvider,
+      chainId,
+      Array.from(targetTokenAddresses),
+    )
+    if (!tokenInfos) return
+    Object.keys(tokenInfos).forEach((address) => {
+      ;(tokenInfos[address] as BasicToken).isLPToken = lpTokens.has(address)
+      ;(tokenInfos[address] as BasicToken).typeAsset =
+        tokenType[address] ?? PoolTypes.OTHER
+    })
+    return tokenInfos
+  })
+}
+
+/**
+ * Fetch basic information about a set of tokens
+ * Does not fetch any user balances
+ */
+async function getTokenInfos(
+  ethCallProvider: MulticallProvider,
+  chainId: ChainId,
+  tokenAddresses: string[], // we assume these are already deduped
+): Promise<BasicTokens | null> {
+  if (!ethCallProvider) {
+    return null
+  }
+  try {
+    const lowercaseTokenAddresses = tokenAddresses.map((address) =>
+      address.toLowerCase(),
+    )
+    const nameCalls = [] as MulticallCall<unknown, string>[]
+    const symbolCalls = [] as MulticallCall<unknown, string>[]
+    const decimalsCalls = [] as MulticallCall<unknown, number>[]
+    lowercaseTokenAddresses.forEach((address) => {
+      const tokenContract = new Contract(
+        address,
+        ERC20_ABI,
+      ) as MulticallContract<Erc20>
+      nameCalls.push(tokenContract.name())
+      symbolCalls.push(tokenContract.symbol())
+      decimalsCalls.push(tokenContract.decimals())
+    })
+
+    const [nameResults, symbolResults, decimalsResults] = await Promise.all([
+      chunkedTryAll(nameCalls, ethCallProvider, 30),
+      chunkedTryAll(symbolCalls, ethCallProvider, 30),
+      chunkedTryAll(decimalsCalls, ethCallProvider, 30),
+    ])
+    const results = lowercaseTokenAddresses.reduce((acc, address, index) => {
+      const name = nameResults[index]
+      const symbol = symbolResults[index]
+      const decimals = decimalsResults[index] // could be 0
+      if (name && symbol && decimals != null) {
```

Comment:
The "else" of this "if" is where the error would be created

Reply:
There are about 3 or more diff ways we can handle failed calls and what logic is inside the useQuery vs what's outside of it. The Promise.all above might throw the error in this case though not too sure. Let's revisit this in a following PR to iron down a good way to construct the queries.